File: themeforest-Fj3VzBCp-fuse-angularjs-material-design-admin-template/fuse/src/app/views/dashboard/mi-asesoria/services/mi-asesoria.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '@/environments/environment';
import { END_POINTS } from '@/app/providers/utils/end-points';

@Injectable({ providedIn: 'root' })
export class MiAsesoriaService {
  private _http = inject(HttpClient);
  private readonly base = environment.url + END_POINTS.asesorias.miAsesoria;
  private readonly solicitudesUrl = environment.url + END_POINTS.asesorias.solicitudes;

  bandeja$(): Observable<any> {
    return this._http.get<any>(this.base);
  }

  /** Descarga un documento (SOLICITUD_ASESORIA | CARTA_ACEPTACION) en formato pdf o docx, como Blob. */
  descargar$(tipo: 'SOLICITUD_ASESORIA' | 'CARTA_ACEPTACION', formato: 'pdf' | 'docx' = 'pdf'): Observable<Blob> {
    return this._http.get(`${this.base}/documentos/${tipo}?formato=${formato}`, { responseType: 'blob' });
  }

  /** Reutiliza el POST de solicitudes: el estudiante solicita a un asesor sugerido. */
  /** @param tipo ASESOR (el principal, uno solo) o COASESOR (opcional, uno solo) */
  solicitar$(docenteId: string, lineaInvestigacionId: string, tituloTentativo?: string,
             tipo: 'ASESOR' | 'COASESOR' = 'ASESOR'): Observable<any> {
    return this._http.post<any>(this.solicitudesUrl, {
      docenteId,
      lineaInvestigacionId,
      tituloTentativo,
      tipo,
    });
  }

  cancelar$(solicitudId: string): Observable<any> {
    return this._http.post<any>(`${this.solicitudesUrl}/${solicitudId}/cancelar`, {});
  }

  /** El estudiante sube su documento firmado (tipo = solicitud | carta). */
  subirFirmado$(tipo: 'solicitud' | 'carta', file: File): Observable<any> {
    const fd = new FormData();
    fd.append('archivo', file, file.name);
    return this._http.post<any>(`${this.base}/documentos/${tipo}/firmado`, fd);
  }

  /** Descarga el dictamen firmado emitido por la secretaría. */
  descargarDictamen$(): Observable<Blob> {
    return this._http.get(`${this.base}/dictamen`, { responseType: 'blob' });
  }

  /** Confirma el envío a Secretaría de los documentos firmados (paso explícito, no automático). */
  enviarDocumentos$(): Observable<any> {
    return this._http.post<any>(`${this.base}/documentos/enviar`, {});
  }
}

File: lamb-upeu-sis/src/main/java/unmsm/edu/pe/tesis/domain/services/MiAsesoriaService.java
package unmsm.edu.pe.tesis.domain.services;

import unmsm.edu.pe.tesis.application.dto.MiAsesoriaResponse;

public interface MiAsesoriaService {

    String TIPO_SOLICITUD = "SOLICITUD_ASESORIA";
    String TIPO_CARTA = "CARTA_ACEPTACION";

    /** Bandeja del estudiante autenticado (tema, sugeridos, estado, documentos). */
    MiAsesoriaResponse bandeja();

    /**
     * PDF de un documento del estudiante autenticado (validación de propiedad implícita:
     * el estudiante se resuelve del token y solo se generan SUS documentos).
     * @param tipo {@link #TIPO_SOLICITUD} o {@link #TIPO_CARTA}
     */
    /** Documento propio (SOLICITUD_ASESORIA | CARTA_ACEPTACION) en formato "pdf" o "docx". */
    byte[] documentoBytes(String tipo, String formato);

    /** El estudiante sube uno de sus documentos firmados (tipo = solicitud | carta). Permite reemplazar. */
    void subirFirmado(String tipo, byte[] contenido, String nombreOriginal, String contentType);

    /** Descarga el dictamen firmado emitido por la secretaría (valida propiedad). */
    unmsm.edu.pe.tesis.application.dto.ArchivoDescargable descargarDictamenEmitido();
}
